airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationDefinitionsHandler.java

```diff
@@ -38,52 +44,128 @@
+      return new DestinationDefinitionRead()
+          .destinationDefinitionId(standardDestinationDefinition.getDestinationDefinitionId())
+          .name(standardDestinationDefinition.getName())
+          .dockerRepository(standardDestinationDefinition.getDockerRepository())
+          .dockerImageTag(standardDestinationDefinition.getDockerImageTag())
+          .documentationUrl(new URI(standardDestinationDefinition.getDocumentationUrl()));
+    } catch (final URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static List<StandardDestinationDefinition> toStandardDestinationDefinitions(Iterator<JsonNode> iter) {
+    Iterable<JsonNode> iterable = () -> iter;
+    var destDefList = new ArrayList<StandardDestinationDefinition>();
+    for (JsonNode e : iterable) {
+      try {
+        var def = mapper.treeToValue(e, StandardDestinationDefinition.class);
+        destDefList.add(def);
+      } catch (JsonProcessingException jsonProcessingException) {
+        jsonProcessingException.printStackTrace();
+        // return 500 if there is an error.
+      }
+    }
+    return destDefList;
   }
 
   public DestinationDefinitionReadList listDestinationDefinitions() throws ConfigNotFoundException, IOException, JsonValidationException {
-    final List<DestinationDefinitionRead> reads = configRepository.listStandardDestinationDefinitions()
+    final List<DestinationDefinitionRead> reads = configRepository
+        .listStandardDestinationDefinitions()
         .stream()
         .map(DestinationDefinitionsHandler::buildDestinationDefinitionRead)
         .collect(Collectors.toList());
 
     return new DestinationDefinitionReadList().destinationDefinitions(reads);
   }
 
+  public DestinationDefinitionReadList listLatestDestinationDefinitions() throws ConfigNotFoundException, IOException, JsonValidationException {
+    final var body = getLatestDestinationsList();
+    final var deserialize = Yamls.deserialize(body);
+    SeedRepository.checkNoDuplicateNames(deserialize.elements());
+    SeedRepository.checkNoDuplicateIds(deserialize.elements(), DEFAULT_DESTINATION_DEFINITION_ID_NAME);
+
+    final var destDefs = toStandardDestinationDefinitions(deserialize.elements());
+    final var reads = destDefs.stream()
+        .map(DestinationDefinitionsHandler::buildDestinationDefinitionRead)
+        .collect(Collectors.toList());
+    return new DestinationDefinitionReadList().destinationDefinitions(reads);
+  }
+
+  private String getLatestDestinationsList() {
+    final var request = HttpRequest.newBuilder(
+        URI.create(DESTINATION_DEFINITION_LIST_LOCATION))
+        .header("accept", "application/json")
+        .build();
+    final var future = client.sendAsync(request, BodyHandlers.ofString());
+    try {
+      final var resp = future.get(1, TimeUnit.SECONDS);
+      return resp.body();
+    } catch (TimeoutException | InterruptedException | ExecutionException e) {
+      throw new RuntimeException(e);
```

Comment:
IIUC, a RTE automatically triggers a http 500 yes?

Reply:
[This](https://github.com/airbytehq/airbyte/blob/master/airbyte-server/src/main/java/io/airbyte/server/errors/UncaughtExceptionMapper.java#L36) tells me it does.